lgc/builder/InOutBuilder.cpp

```diff
@@ -739,9 +739,10 @@ Instruction *InOutBuilder::CreateWriteXfbOutput(Value *valueToWrite, bool isBuil
   unsigned streamId = outputInfo.hasStreamId() ? outputInfo.getStreamId() : 0;
   assert(xfbBuffer < MaxTransformFeedbackBuffers);
   assert(streamId < MaxGsStreams);
-  resUsage->inOutUsage.xfbStrides[xfbBuffer] = xfbStride;
-  resUsage->inOutUsage.enableXfb = true;
-  resUsage->inOutUsage.streamXfbBuffers[streamId] |= 1 << xfbBuffer;
+  auto &xfbStrides = getPipelineState()->getXfbBufferStrides();
+  auto &streamXfbBuffers = getPipelineState()->getStreamXfbBuffers();
+  xfbStrides[xfbBuffer] = xfbStride;
+  streamXfbBuffers[streamId] |= 1 << xfbBuffer;
```

Comment:
This shouldn't be needed anymore.

Reply:
Without updating `xfbstrides` and `streamXfbBuffers`, the block array cases will crash. 
Regression on _dEQP-VK.transform_feedback.fuzz.2_level_struct_array.instance_array_vertex_ [blockArray.zip](https://github.com/GPUOpen-Drivers/llpc/files/10051905/blockArray.zip)

```
#version 450
struct S
{
	highp uvec3 a; // unassigned
	mediump mat2 b;
};
layout(location = 0, xfb_buffer = 0, xfb_offset = 0, xfb_stride = 124) out Block // sizeInBytes=124 sizeInLocations=14
{
	mediump int u;
	S s[2][2];
	mediump vec2 v;
} block[2];
void main (void)
{
	block[0].u = -89;
	block[0].s[0][0].b = mat2(30.0, -70.0, -60.0, 72.0);
	block[0].s[0][1].b = mat2(-102.0, -100.0, 84.0, -31.0);
	block[0].s[1][0].b = mat2(68.0, 104.0, 14.0, 83.0);
	block[0].s[1][1].b = mat2(-117.0, 75.0, 12.0, -18.0);
	block[0].v = vec2(-101.0f, 100.0f);
	block[1].u = -34;
	block[1].s[0][0].b = mat2(-35.0, -43.0, 48.0, 4.0);
	block[1].s[0][1].b = mat2(-60.0, 37.0, 52.0, 98.0);
	block[1].s[1][0].b = mat2(-80.0, -62.0, -54.0, -79.0);
	block[1].s[1][1].b = mat2(3.0, -97.0, -118.0, -108.0);
	block[1].v = vec2(-120.0f, 48.0f);
}
```
In SPIRVReader, we can only get `XfbBuffer 0` used. After `lowerOutput`, `block[0]` is streamed out `XbfBuffer 0` and `block[1]` is streamed out `XfbBuffer 1`. See the effect of parameter `xfbBufferAdjust` of `addCallInstForOutputExport`.

Reply:
Interesting, thank you, I hadn't realized that.

Though now I'm curious: what is the desired behavior if `block[1]` is never written in the shader? Shouldn't the write offset of the corresponding xfbBuffer still be incremented, and doesn't that also require capturing the information in SPIRVReader?